source/common/upstream/upstream_impl.cc

```diff
@@ -102,6 +133,16 @@ HostImpl::createConnection(Event::Dispatcher& dispatcher, const ClusterInfo& clu
   } else {
     cluster_options = options;
   }
+  if (cluster.features() & ClusterInfo::Features::USE_TCP_KEEPALIVE) {
+    if (!cluster_options) {
```

Comment:
Please clean up with duplicate logic from the FREEBIND case above. This is hard to follow.